File: src/main/java/com/service/model/Service.java
package com.service.model;

import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name="service")
public class Service {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "Id")
    private int Id;

    @Column(name = "expert_name")
    private String expertName;

    @Column(name = "title")
    private String title;

    @Column(name = "professional_type")
    private String professionType;

    @Column(name = "text")
    private String description;

    @Column(name = "service_type")
    private String serviceType;

    @Column(name = "subscribe_time")
    @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private Date time;

    public String getExpertName() {
        return expertName;
    }

    public void setExpertName(String expertName) {
        this.expertName = expertName;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getProfessionType() {
        return professionType;
    }

    public void setProfessionType(String professionType) {
        this.professionType = professionType;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getServiceType() {
        return serviceType;
    }

    public void setServiceType(String serviceType) {
        this.serviceType = serviceType;
    }

    public Date getTime() {
        return time;
    }

    public void setTime(Date time) {
        this.time = time;
    }

    public Date getDeadline() {
        return deadline;
    }

    public void setDeadline(Date deadline) {
        this.deadline = deadline;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    @Column(name = "deadline")
    @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private Date deadline;

    @Column(name = "subscribe_address")
    private String address;

    public int getId() {
        return Id;
    }

    public void setId(int id) {
        Id = id;
    }


}

File: src/main/resources/static/js/service.js
// 验证用户输入

function verifyServiceValue(service) {
    if (!service.title || service.title === '') {
        return {
            success: false,
            message: '标题不能为空',
            data: null
        };
    }

    if (!service.address || service.address === '') {
        return {
            success: false,
            message: '地址不能为空',
            data: null
        };
    }

    if (!service.time || service.time === '') {
        return {
            success: false,
            message: '预约时间不能为空',
            data: null
        };
    }

    if (!service.deadline || service.deadline === '') {
        return {
            success: false,
            message: '截止时间不能为空',
            data: null
        };
    }

    if (!service.serviceType || service.serviceType === '') {
        return {
            success: false,
            message: '服务类型不能为空',
            data: null
        };
    }

    if (!service.professionType || service.professionType === '') {
        return {
            success: false,
            message: '专业类型不能为空',
            data: null
        };
    }

    if (!service.description || service.description === '') {
        return {
            success: false,
            message: '正文不能为空',
            data: null
        };
    }

    return {
        success: true,
        message: '创建服务成功',
        data: null
    };
}